drivers/gpio/gpio_silabs.c

```diff
@@ -236,30 +235,34 @@ static int gpio_silabs_pin_interrupt_configure(const struct device *dev, gpio_pi
 					       enum gpio_int_mode mode, enum gpio_int_trig trig)
 {
 	const struct gpio_silabs_port_config *config = dev->config;
-	struct gpio_silabs_port_data *data = dev->data;
-	struct gpio_silabs_common_data *common = data->common_dev->data;
-	sl_gpio_t gpio = {.port = config->gpio_index, .pin = pin};
+	struct gpio_silabs_common_data *common = config->common_dev->data;
+	sl_gpio_t gpio = {
+		.port = config->gpio_index,
+		.pin = pin
+	};
 	int32_t int_no = SL_GPIO_INTERRUPT_UNAVAILABLE;
 	sl_gpio_interrupt_flag_t flag = SL_GPIO_INTERRUPT_RISING_FALLING_EDGE;
+	uint32_t enabled_interrupts;
 
 	if (mode == GPIO_INT_MODE_LEVEL) {
 		LOG_ERR("Level interrupt not supported on pin %u", pin);
 		return -ENOTSUP;
 	}
 
+	enabled_interrupts = sl_hal_gpio_get_enabled_interrupts();
+
 	if (mode == GPIO_INT_MODE_DISABLED) {
 		ARRAY_FOR_EACH(common->irq_pin_map, i) {
-			if ((data->int_enabled_mask & BIT(i)) && (common->irq_pin_map[i] == pin)) {
+			if ((enabled_interrupts & BIT(i)) && (common->irq_pin_map[i] == pin)) {
```

Comment:
In fact `common->irq_pin_map[i] == pin` is sufficient. If the interrupt is not enabled, it means the data in inconsistent and we have a bug. Probably an assert is sufficient.